src/components/TodoItem.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useContext } from 'react';
+import classNames from 'classnames';
+import { TodosContext } from '../store';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  item: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ item }) => {
+  const { dispatch } = useContext(TodosContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(item.title);
+
+  const classes = classNames({
+    completed: item.completed,
+    editing: isEditing,
+  });
+
+  const handleBlur = () => {
+    setIsEditing(false);
+
+    dispatch({
+      type: 'UPDATE_TODO_TITLE',
+      payload: { id: item.id, title: editedTitle },
+    });
+  };
+
+  const handleKeyUp = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      setIsEditing(false);
+
+      dispatch({
+        type: 'UPDATE_TODO_TITLE',
+        payload: { id: item.id, title: editedTitle },
+      });
+    }
+
+    if (e.key === 'Escape') {
+      setIsEditing(false);
+    }
+  };
+
+  return (
+    <li
+      className={classes}
+      onBlur={handleBlur}
+      onDoubleClick={() => setIsEditing(true)}
+    >
+      <div className="view">
+        <input
+          type="checkbox"
+          className="toggle"
+          checked={item.completed}
+          onChange={() => dispatch({ type: 'TOGGLE_TODO', payload: item })}
```

Comment:
Consider will be better if you move this logic from the jsx to the helper function and use it everywhere